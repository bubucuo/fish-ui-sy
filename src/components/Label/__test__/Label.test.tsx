import { render } from "@testing-library/react";
import { Label } from "fish-ui-sy";

describe("Label", () => {
  it("renders a default state", () => {
    const result = render(<Label>Default Label</Label>);
    expect(result.container).toMatchSnapshot();
  });

  it("renders a custom asterisk", () => {
    const result = render(
      <Label required="foo" data-testid="label-id">
        I'm a label
      </Label>
    );
    const label = result.getByTestId("label-id");
    const requiredSlot = label.children[0];
    if (requiredSlot) {
      expect(requiredSlot.innerHTML).toBe("foo");
    } else {
      throw Error("Custom required text was not rendered");
    }
    expect(result.baseElement).toMatchSnapshot();
  });
});
