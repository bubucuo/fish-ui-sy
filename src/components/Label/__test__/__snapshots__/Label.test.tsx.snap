// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > renders a custom asterisk 1`] = `
<body>
  <div>
    <label
      aria-hidden="true"
      class="fish-ui-Label ___1yevxz2_yljzsa0 fk6fouc f19n0e5 fkhj508 f1i3iumi"
      data-testid="label-id"
    >
      I'm a label
      <span
        class="fish-ui-Label__required ___di7pjm0_79xo450 f1whyuy6 fruq291"
      >
        foo
      </span>
    </label>
  </div>
</body>
`;

exports[`Label > renders a default state 1`] = `
<div>
  <label
    aria-hidden="true"
    class="fish-ui-Label ___1yevxz2_yljzsa0 fk6fouc f19n0e5 fkhj508 f1i3iumi"
  >
    Default Label
  </label>
</div>
`;
